api/app/controllers/api/v1/applications_controller.rb

```diff
@@ -84,13 +107,21 @@ def matching_data_update_params
         end
         
         def applications_by_session
-            return Application.order(:id).select do |entry|
+            return all_applications.select do |entry|
                 entry[:session_id] == params[:session_id].to_i
             end
         end
 
+        def all_applications
+            exclusion = [:id, :created_at, :updated_at, :application_id]
+            return Application.order(:id).map do |entry|
+                matching = ApplicantDataForMatching.find_by(application_id: entry[:id])
```

Comment:
it might be better to use a join here so we don't need to make a db call for each Application, but we don't need to worry about it for now.